packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1639,7 +1646,24 @@ function updateSuspenseComponent(
       primaryChildFragment.sibling = fallbackChildFragment;
       // Skip the primary children, and continue working on the
       // fallback children.
-      workInProgress.memoizedState = SUSPENDED_MARKER;
+
+      if (workInProgress.effectTag & SoftCapture) {
+        // don't clear the effectTag yet, it'll be read in completeWork
+        workInProgress.effectTag &= ~SoftCapture;
+        const avoidedRenderExpiration = renderExpirationTime - 1; // verify/rename
+        // this means there's work to be done on this fiber "in the future"
+        workInProgress.memoizedState = {
+          dehydrated: null,
+          retryTime: NoWork,
+          didAvoidRenderTime: avoidedRenderExpiration,
+        };
+
+        workInProgress.expirationTime = avoidedRenderExpiration;
+      } else {
+        // workInProgress.effectTag & DidCapture is truthy
```

Comment:
This comment is a bit too specific. It's just describing the technical details (also, we shouldn't actually be comparing truthiness but compare against zero so that's an oversight). You might want to describe it more as a higher level concept.

However, I think this points out a bug though. What if both are set? In completeWork, DidCapture wins, but in beginWork SoftCapture wins. That seems like it could cause havoc.

Reply:
rewrote for DidCapture to win first

Reply:
scratch that, it fails tests. reverting and investigating.